### ISSUE TITLE
possible error in quota handling?

### ISSUE THREAD

**Issue by ericblade** [ericblade/mws-advanced#114]
just received this on my service, a few times, i'm not sure if it's related to the new quota handling code, or if it's something inside my app service.

````
(node:13300) TimeoutOverflowWarning: Infinity does not fit into a 32-bit signed integer.
````

**Comment by pwrnrd** [comment]
Are you running your app on #113?

**Comment by ericblade** [comment]
service is on node 9.6.1 right now, and i'm reasonably certain it's using the most recent commit here . . . i haven't replicated the problem, yet, but it does seem that somewhere someone has done a setTimeout with Infinity, which doesn't sound like anything i'm doing on the app that uses mws-advanced... but i need to go over it with a comb to make sure.

**Comment by pwrnrd** [comment]
There is also a timer in `Queue.complete()` which might return infinity: 

````
 const time = (((60 / this.restoreRate) || 1) * 1000 * this.maxInFlight) + 250;
 if (this.resetDrainTimeout) {
       clearTimeout(this.resetDrainTimeout);
  }
  this.resetDrainTimeout = setTimeout(() => { this.singleDrain = false; }, time);
````

**Comment by ericblade** [comment]
Thanks for finding that, I had a feeling that was something involving time and division.  I honestly thought node would throw a runtime error for division by zero rather than returning Infinity, though. heh.

looks to me like it should be
```` const time = (60 / (this.restoreRate || 1) * 1000 * this.maxInFlight) + 250;````
or something like that.
i wonder what i'm calling that doesn't have the throttle data filled in.

**Comment by pwrnrd** [comment]
> i wonder what i'm calling that doesn't have the throttle data filled in.
Exactly! I was thinking about setting the default value to 1. But than I though, perhaps we should throw an error. Personally I have a preference for an error, because I don't expect that we'll ever have to deal with an MWS endpoint which has no restore rate...

**Comment by ericblade** [comment]
well, there's a bunch of quota rate that just has never had the data filled in, and i'm not sure if that's because documentation was lacking, or i just didn't get to it, because the calls weren't useful to my purposes at the time.  i might be using a mws call that doesn't have a helper wrapper thing.
